### PR TITLE
gitkraken: fix broken interpreter after recent update

### PR DESCRIPTION
**Pull request by toelke**
###### Description of changes

This hopes to fix #206240

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sheepforce** [comment]
I can confirm that this works as expected, now :)

**Review by sheepforce — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

File: pkgs/applications/version-management/gitkraken/default.nix [hunk 1]
```diff
@@ -124,6 +124,9 @@ let
       pushd $out/share/${pname}
       patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" ${pname}
 
+      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" chrome-sandbox
+      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" chrome_crashpad_handler
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/gitkraken/default.nix:128]:
```suggestion
      for file in ${pname} chrome-sandbox chrome_crashpad_handler; do
        patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $file
      done
```

**Comment by rhoriguchi** [comment]
I just tested it and it still fails. What parameters did you use for `strace`?

**Comment by toelke** [comment]
`strace -f gitkraken`

What do `ldd $(which gitkraken)` and `file $(which gitkraken)` say?

**Comment by rhoriguchi** [comment]
```console
> ldd $(which gitkraken)
	linux-vdso.so.1 (0x00007fffe3b8c000)
	libc.so.6 => /nix/store/ayfr5l52xkqqjn3n4h9jfacgnchz1z7s-glibc-2.35-224/lib/libc.so.6 (0x00007f37be686000)
	/nix/store/ayfr5l52xkqqjn3n4h9jfacgnchz1z7s-glibc-2.35-224/lib/ld-linux-x86-64.so.2 => /nix/store/ayfr5l52xkqqjn3n4h9jfacgnchz1z7s-glibc-2.35-224/lib64/ld-linux-x86-64.so.2 (0x00007f37be891000)

> file $(which gitkraken)
/run/current-system/sw/bin/gitkraken: symbolic link to /nix/store/zkcn5hb5xf5gfkffg5yg0j4918a8cq47-gitkraken-9.0.0/bin/gitkraken
```

`strace -f gitkraken`
[trace.txt](https://github.com/NixOS/nixpkgs/files/10421372/trace.txt)

**Comment by toelke** [comment]
Hm, I forgot that `file` would not  follow symlinks by default :-D

Can you also do `file -L $(which gitkraken)`?

**Comment by rhoriguchi** [comment]
```console
> file -L $(which gitkraken)

/run/current-system/sw/bin/gitkraken: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /nix/store/ayfr5l52xkqqjn3n4h9jfacgnchz1z7s-glibc-2.35-224/lib/ld-linux-x86-64.so.2, for GNU/Linux 3.10.0, not stripped
```

**Comment by toelke** [comment]
I think this might be a different problem than the one I fixed in this PR:

`[2904161:0115/203658.856659:FATAL:gpu_data_manager_impl_private.cc(448)] GPU process isn't usable. Goodbye.`

Does it display anything at all? Does `gitkraken --disable-gpu` work?

**Review by rhoriguchi — APPROVED** [review]

**Comment by rhoriguchi** [comment]
> I think this might be a different problem than the one I fixed in this PR:
> 
> `[2904161:0115/203658.856659:FATAL:gpu_data_manager_impl_private.cc(448)] GPU process isn't usable. Goodbye.`
> 
> Does it display anything at all? Does `gitkraken --disable-gpu` work?

`--disable-gpu` fixes this issue. Thank you for the help.

**Comment by hacker1024** [comment]
For the impatient:

```nix
(gitkraken.overrideAttrs ({ pname, postFixup, ... }: {
  postFixup = ''
    ${postFixup}
    pushd $out/share/${pname}
    for file in chrome-sandbox chrome_crashpad_handler; do
      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $file
    done
    popd
  '';
}))
```

**Pull request merged**